hbase-server/src/main/java/org/apache/hadoop/hbase/replication/master/ReplicationLogCleaner.java

```diff
@@ -192,6 +192,14 @@ public Iterable<FileStatus> getDeletableFiles(Iterable<FileStatus> files) {
     if (this.getConf() == null) {
       return files;
     }
+    try {
+      if (!rpm.getQueueStorage().hasData()) {
+        return files;
+      }
+    } catch (ReplicationException e) {
+      LOG.error("Error occurred while executing queueStorage.hasData()", e);
+      return files;
```

Comment:
Should return empty here for safety?

Reply:
Thank you very much for your suggestion. It has been modified